examples/chol.dx

```diff
@@ -0,0 +1,96 @@
+' Cholesky Factorization
+https://en.wikipedia.org/wiki/Cholesky_decomposition
+
+chol :: (n=>n=>Real) -> (n=>n=>Real)
+chol x = ..
+  inUpper:: A m::Ix. m -> m -> Bool
+  inUpper i j = (asint i) < (asint j)
+  inLower:: A m::Ix. m -> m -> Bool
+  inLower i j = (asint i) > (asint j)
+  ..
+  diagOp r i = sqrt (x.i.i - sum for k::n. select ( inLower k i) (sq r.i.k) 0.0)
+  lowerOp r i j = 
+    innersum = sum for k. ..
+      select (inLower j k) (r.i.k * r.j.k) 0.0
+    (x.i.j - innersum) / r.j.j
+  ..
+  fs = ..
+  for outerrow::n. ..
+    lam r.
+    for i::n. ..
+      select (inUpper i outerrow) ..
+        (for j::n. r.i.j) .. -- copy 
```

Comment:
If this doesn't optimize out, then it is an unfortunate increase in time complexity of the algorithm  over the mutating version.